Source/Tests/LinuxSyscalls/SignalDelegator.cpp

```diff
@@ -590,21 +600,19 @@ namespace FEX::HLE {
 
       // Now actually set the host mask
       // This will hide from the guest that we are not actually setting all of the masks it wants
-      sigset_t HostSet{};
-      sigemptyset(&HostSet);
-
+      uint64_t HostSet{};
       for (size_t i = 0; i < MAX_SIGNALS; ++i) {
         if (HostHandlers[i + 1].Required.load(std::memory_order_relaxed)) {
           // If it is a required host signal then we can't mask it
           continue;
         }
 
         if (ThreadData.CurrentSignalMask.Val & (1ULL << i)) {
-          sigaddset(&HostSet, i + 1);
+          HostSet |= (1ULL << 1);
         }
       }
 
-      pthread_sigmask(SIG_SETMASK, &HostSet, nullptr);
+      ::syscall(SYS_rt_sigprocmask, SIG_SETMASK, &HostSet, nullptr);
```

Comment:
oop, this needs a final argument of sigsetsize and to handle 32-bit sigprocmask sigsetsize.